dlib/matrix/matrix_fft_abstract.h

```diff
@@ -231,6 +231,151 @@ namespace dlib
                   inverse transformation.  
     !*/
 
+// ----------------------------------------------------------------------------------------
+
+    template <typename EXP>
+    matrix<complex_type> stft (
+        const matrix_exp<EXP>& signal,
+        WindowType w,
```

Comment:
Can you put the `WindowType` enum in this file too?  The abstract docs file shouldn't refer to anything that isn't in the docs.  Since otherwise people can't know about it by reading just the docs.